source/common/http/http1/conn_pool.cc

```diff
@@ -268,6 +268,9 @@ ConnPoolImpl::ActiveClient::ActiveClient(ConnPoolImpl& parent)
   parent_.conn_connect_ms_ =
       parent_.host_->cluster().stats().upstream_cx_connect_ms_.allocateSpan();
   Upstream::Host::CreateConnectionData data = parent_.host_->createConnection(parent_.dispatcher_);
+  if (data.connection_->state() != Network::Connection::State::Open) {
```

Comment:
IMO this approach is a bit fragile in that we don't really know that this is a bind error. Perhaps something else might change in the future? It also has the downside of not covering other cases such as tcp_proxy, redis, etc.

I see two options here:
1) Rename/extend `setBufferStats` (https://github.com/lyft/envoy/blob/master/include/envoy/network/connection.h#L147) to include other stats also. Then a call could pass in a bind failure stat if they want.
2) Make `createConnection()` take a struct with optional stats in it, which will force people to plumb things through as needed (though this would potentially require changes to more callsites).

Thoughts?

Reply:
We could avoid the first concern by just adding an ASSERT in the connection that if the connection isn't connected there is a bind failure, and adding a comment pointer to anyone changing the code.  Agree, the second, that this misses tcp_proxy, is more of a problem though.

setting buffer stats on creation looks super duper messy since for example in the case of the tcp proxy filter, the connection is created prior to the read callbacks which is where buffer stats are set  

I'd be inclined to go with renaming setBufferStats to be more general, and then somewhat unintuitively incrementing the bind error where we raise the error onFileEvent with a comment that we defer until the caller has a chance to set [buffer] stats.  If that doesn't sound too hacky to you I'll take a whack at that tomorrow

Reply:
I think I'm fine with that unless you can think of a better option. I would rename to something like `setConnectionStats` or `setStats` or something.

Reply:
Done!